geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java

```diff
@@ -54,277 +45,163 @@
       final String hostedLocator = hostedLocators.get(member).iterator().next();
-      assertTrue(locators + " does not contain " + hostedLocator, locators.contains(hostedLocator));
+      assertThat(locators).contains(hostedLocator);
     }
+  }
 
-    // validate fix for #46324
-    for (int whichvm = 0; whichvm < 4; whichvm++) {
-      getHost(0).getVM(whichvm).invoke(new SerializableRunnable() {
-        @Override
-        public void run() {
-          final ClusterDistributionManager dm =
-              (ClusterDistributionManager) InternalDistributedSystem.getAnyInstance()
-                  .getDistributionManager();
-          final InternalDistributedMember self = dm.getDistributionManagerId();
-
-          final Set<InternalDistributedMember> locatorIds = dm.getLocatorDistributionManagerIds();
-          assertTrue(locatorIds.contains(self));
+  public static InetAddress getIPAddress() {
+    return Boolean.getBoolean("java.net.preferIPv6Addresses") ? getIPv6Address() : getIPv4Address();
+  }
 
-          final Map<InternalDistributedMember, Collection<String>> hostedLocators =
-              dm.getAllHostedLocators();
-          assertTrue(
-              "hit bug #46324: " + hostedLocators + " is missing "
-                  + InternalLocator.getLocatorStrings() + " for " + self,
-              hostedLocators.containsKey(self));
-        }
-      });
+  protected static InetAddress getIPv4Address() {
+    InetAddress host = null;
+    try {
+      host = InetAddress.getLocalHost();
+      if (host instanceof Inet4Address && !host.isLinkLocalAddress() && !host.isLoopbackAddress()) {
+        return host;
+      }
+    } catch (UnknownHostException e) {
+      String s = "Local host not found";
+      throw new RuntimeException(s, e);
     }
-
-    // validation with locators
-    for (int whichvm = 0; whichvm < 4; whichvm++) {
-      getHost(0).getVM(whichvm).invoke(new SerializableRunnable() {
-        @Override
-        public void run() {
-          final ClusterDistributionManager dm =
-              (ClusterDistributionManager) InternalDistributedSystem.getAnyInstance()
-                  .getDistributionManager();
-
-          final Set<InternalDistributedMember> locatorIds = dm.getLocatorDistributionManagerIds();
-          assertEquals(5, locatorIds.size());
-
-          final Map<InternalDistributedMember, Collection<String>> hostedLocators =
-              dm.getAllHostedLocators();
-          assertTrue(!hostedLocators.isEmpty());
-          assertEquals(5, hostedLocators.size());
-
-          for (InternalDistributedMember member : hostedLocators.keySet()) {
-            assertEquals(1, hostedLocators.get(member).size());
-            final String hostedLocator = hostedLocators.get(member).iterator().next();
-            assertTrue(locators + " does not contain " + hostedLocator,
-                locators.contains(hostedLocator));
+    try {
+      Enumeration i = NetworkInterface.getNetworkInterfaces();
```

Comment:
declare as `Enumeration<NetworkInterface> i` to eliminate the cast on line 156.

Reply:
All review comments have been addressed